### PR TITLE
Port over android driver key event tests

### PR DESCRIPTION
**Pull request by imurchie**
Direct port of `appium-android-driver` tests for key event functionality, to complete https://github.com/appium/appium-uiautomator2-driver/pull/85

### CODE REVIEW

File: test/functional/commands/keyboard/keyboard-e2e-specs.js [hunk 1]
```diff
@@ -49,6 +60,55 @@ async function runTextEditTest (driver, testText, keys = false) {
...
+  let runTest = async function () {
+    await driver.pressKeyCode(29, 193);
+    let el = await getElement(driver, TEXTVIEW_CLASS);
+    return await driver.getText(el);
+  };
+
+  await clearKeyEvents(driver);
+
+  let text = await runTest();
+  if (text === '') {
```
Comment (mykola-mokhnach) [inline test/functional/commands/keyboard/keyboard-e2e-specs.js]: can be replaced with `if (!text)`

File: test/functional/commands/keyboard/keyboard-e2e-specs.js [hunk 2]
```diff
@@ -49,6 +60,55 @@ async function runTextEditTest (driver, testText, keys = false) {
...
+async function clearKeyEvents (driver) {
+  let el = await getElement(driver, BUTTON_CLASS);
+  driver.click(el);
+
+  // wait a moment for the clearing to occur, lest we too quickly try to enter more text
+  await B.delay(500);
+}
+
+async function runCombinationKeyEventTest (driver) {
+  let runTest = async function () {
```
Comment (mykola-mokhnach) [inline test/functional/commands/keyboard/keyboard-e2e-specs.js]: should it be a lambda?

File: test/functional/commands/keyboard/keyboard-e2e-specs.js [hunk 3]
```diff
@@ -49,6 +60,55 @@ async function runTextEditTest (driver, testText, keys = false) {
...
+async function runKeyEventTest (driver) {
+  let runTest = async function () {
+    await driver.pressKeyCode(82);
+    let el = await getElement(driver, TEXTVIEW_CLASS);
+    return await driver.getText(el);
+  };
+
+  await clearKeyEvents(driver);
+
+  let text = await runTest();
```
Comment (mykola-mokhnach) [inline test/functional/commands/keyboard/keyboard-e2e-specs.js]: I see code duplication here

File: test/functional/commands/keyboard/keyboard-e2e-specs.js (hunk 2 above)
Comment (imurchie) [inline test/functional/commands/keyboard/keyboard-e2e-specs.js]: In what sense?

Comment (mykola-mokhnach) [inline test/functional/commands/keyboard/keyboard-e2e-specs.js]: `let runTest = async function () {` -> `let runTest = async () => {`

Comment (imurchie) [inline test/functional/commands/keyboard/keyboard-e2e-specs.js]: Ah. There is no gain from that, here. The function isn't relying on any scoping.

File: test/functional/commands/keyboard/keyboard-e2e-specs.js (hunk 3 above)
Comment (imurchie) [inline test/functional/commands/keyboard/keyboard-e2e-specs.js]: Gone!

Comment (imurchie) [inline test/functional/commands/keyboard/keyboard-e2e-specs.js]: Though now I really should go and do it in the android driver, too. Alas.

Comment (mykola-mokhnach) [inline test/functional/commands/keyboard/keyboard-e2e-specs.js]: 👿

**Pull request merged**